lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -192,22 +196,37 @@ export default class LockMonitor {
     }
 
     // If we're here then it means that we have saved some information about a lock
-    // which is confirmed to be on the blockchain. Let's resolve it to make sure that
-    // we have all the information.
+    // which is at least broadcasted to blockchain. Let's resolve it.
     const lastLockIdentifier: LockIdentifier = {
       transactionId: lastSavedLock.transactionId,
       redeemScriptAsHex: lastSavedLock.redeemScriptAsHex
     };
 
-    const currentValueTimeLock = await this.lockResolver.resolveLockIdentifierAndThrowOnError(lastLockIdentifier);
+    try {
+      const currentValueTimeLock = await this.lockResolver.resolveLockIdentifierAndThrowOnError(lastLockIdentifier);
 
-    console.info(`Found a valid current lock: ${JSON.stringify(currentValueTimeLock)}`);
+      console.info(`Found a valid current lock: ${JSON.stringify(currentValueTimeLock)}`);
 
-    return {
-      currentValueTimeLock: currentValueTimeLock,
-      latestSavedLockInfo: lastSavedLock,
-      status: LockStatus.Confirmed
-    };
+      return {
+        currentValueTimeLock: currentValueTimeLock,
+        latestSavedLockInfo: lastSavedLock,
+        status: LockStatus.Confirmed
+      };
+
+    } catch (e) {
+
+      if (e instanceof BitcoinError && e.code === ErrorCode.LockResolverTransactionNotConfirmed) {
+        // This means that the transaction was broadcasted but hasn't been written on the blockchain yet.
+        return {
+          currentValueTimeLock: undefined,
+          latestSavedLockInfo: lastSavedLock,
+          status: LockStatus.Pending

```

Comment:
No action required, more of a discussion: `currentValueTimeLock` being `undefined` when `LockStatus` is `Pending` is throwing me off, I see the logic being the lock is not "current" unless it is "confirmed and not expired", in which case we can also consider naming it `activeValueTimeLock`.

Reply:
I was thinking on the same lines actually. Let me change the name and add some comments around the field to be more clear about the usage.